codex/stores/maintenance.nim

```diff
@@ -0,0 +1,93 @@
+## Nim-Codex
+## Copyright (c) 2023 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+## Store maintenance module
+## Looks for and removes expired blocks from blockstores.
+
+import pkg/chronos
+import pkg/chronicles
+import pkg/questionable
+import pkg/questionable/results
+
+import codex/stores/repostore
+import codex/utils/timer
+import codex/clock
+import codex/systemclock
+
+type
+  BlockMaintainer* = ref object of RootObj
+    repoStore: RepoStore
+    interval: Duration
+    timer: Timer
+    clock: Clock
+    numberOfBlocksPerInterval: int
+    offset: int
+
+proc new*(T: type BlockMaintainer,
+    repoStore: RepoStore,
+    interval: Duration,
+    timer = Timer.new(),
+    clock: Clock = SystemClock.new(),
+    numberOfBlocksPerInterval = 100
+    ): T =
+  T(
+    repoStore: repoStore,
+    interval: interval,
+    timer: timer,
+    clock: clock,
+    numberOfBlocksPerInterval: numberOfBlocksPerInterval,
+    offset: 0
+  )
+
+proc runBlockCheck(self: BlockMaintainer): Future[void] {.async.} =
+  var blockCidsToDelete = newSeq[Cid](0)
```

Comment:
I'm not sure if this makes much sence, let me explain - the point of an iterator is that we don't have to consume all items coming from the source before having to process them, essentially, an `iterator` is a streaming primmitive, what's being done here invalidates the purpose of the `iterator`. I would suggest we rework this so that we delete expired records as they are being read from the `iterator`.

Reply:
I understand that using the seq[Cid] goes against the way iterators work. I used it here to avoid the classic 'modifying a collection during iteration'. I would imagine the underlying datastore at the moment can handle it, but I didn't want to build this assumption into the code.
I considered the max memory use of seq[Cid] and I figured it is probably harmless. What do you think?

Reply:
The underlying store should definitely support this functionality, sqlite does and most other DB or KV stores support it as well, the only other thing could be the `fsstore`, but it is implemented to be safe there as well. In other words, we should probably make it a requirement for stores to support retrive-modify operations with cursors/iterators.

Reply:
All right, easy fix :D